### ISSUE TITLE
LibWeb: Overflow clipping does not work for `position: absolute` child of `position: absolute` div

### ISSUE THREAD

**Issue by MacDue** [SerenityOS/serenity#16039]
Minimal reproducer: 
```
<style>
  #clip {
    position: absolute;
    width: 100px;
    height: 200px;
    border-radius: 50%;
    overflow: hidden;
  }

  #child {
    position: absolute;
    height: 100%;
    width: 100%;
    background-color: red;
  }
</style>
<div id=clip>
    <div id=child>
    </div>
</div>
```
Reduced from: https://diana-adrianne.com/purecss-francine/ (looks like fixing this could fix quite a few issues there).

| LibWeb (Incorrect)                                                                                              | Chrome (Correct)                                                                                                |
|-----------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|
| ![image](https://user-images.githubusercontent.com/11597044/201478470-ce5dfa7b-27dd-4742-a4ac-7a859e39575d.png) | ![image](https://user-images.githubusercontent.com/11597044/201478484-f824a11c-444e-4d30-a763-defa35fbe92c.png) |

**Comment by not-a-skeleton** [comment]
Looks like this ended up being resolved by a later commit in #16035. In commit 16f2904 LibWeb's rendering of the reduced test case now matches the "Chrome (Correct)" screenshot.

**Comment by MacDue** [comment]
@not-a-skeleton Yep, sorry that this was not closed then.

Fixed by #16035
